helm/resource_release_test.go

```diff
@@ -918,6 +971,38 @@ func TestGetValuesString(t *testing.T) {
 	}
 }
 
+func TestUseChartVersion(t *testing.T) {
+
+	type test struct {
+		chartPath       string
+		repositoryURL   string
+		useChartVersion bool
+	}
+
+	tests := []test{
+		// when chart is a local directory
+		{chartPath: "./testdata/charts/test-chart", repositoryURL: "", useChartVersion: true},
+		// when the repo is a local directory
+		{chartPath: "testchart", repositoryURL: "./testdata/charts", useChartVersion: true},
+		// when the repo is a repository URL
+		{chartPath: "", repositoryURL: "https://charts.bitnami.com/bitnami", useChartVersion: false},
+		// when chartPath is chart name and repo is repository URL
+		{chartPath: "redis", repositoryURL: "https://charts.bitnami.com/bitnami", useChartVersion: false},
+		// when the chart is a URL to an .tgz file
+		{chartPath: "https://charts.bitnami.com/bitnami/redis-10.7.16.tgz", repositoryURL: "", useChartVersion: true},
+		// when the repo is an OCI registry
+		{chartPath: "redis", repositoryURL: "oci://registry-1.docker.io/bitnamicharts", useChartVersion: false},
+		// when the chart is a URL to an OCI registry
+		{chartPath: "oci://registry-1.docker.io/bitnamicharts/redis", repositoryURL: "", useChartVersion: true},
```

Comment:
This should be `false` — it's a path to an OCI registry, not a local file or URL to an archive.

Reply:
It was set to `true` since we are specifically getting `oci://registry-1.docker.io/bitnamicharts/redis` chart but no version is set. So if the user wants to use a specific version they would need to set it in `.version` and if none is set it will grab the latest release.

I made some changes to the test and to `useChartVersion` since the tests were before the name change so the logic didn't make sense now when switching from `IsLocalChart` -> `useChartVersion`

Reply:
this came due to name confusion, I've fixed now to match the logic that you were mentioning. Thanks for catching this.